fixed file permissions for admin.kubeconfig and execution of oadm

PI-Victor opened the pull request:
i've noticed admin.kubeconfig needs rw permissions for an unprivileged user to be able to create a new registry or a new project, and also added execution for for oadm, export PATH is the step below that and it doesn't resolve so people might find it confusing.

mfojtik commented on CONTRIBUTING.adoc:
```diff
@@ -92,8 +92,8 @@ NOTE: Replace "linux/amd64" with the appropriate value for your platform/archite
-        $ sudo chmod +r openshift.local.config/master/admin.kubeconfig
-        $ oadm registry --create --credentials=openshift.local.config/master/openshift-registry.kubeconfig --config=openshift.local.config/master/admin.kubeconfig
+        $ sudo chmod a+rw openshift.local.config/master/admin.kubeconfig
+        $ ./oadm registry --create --credentials=openshift.local.config/master/openshift-registry.kubeconfig --config=openshift.local.config/master/admin.kubeconfig
```
why is `./` needed here?

PI-Victor replied:
because you didn't add the openshift binaries to path yet, am i missing something?

kargakis replied:
can you do that instead? see https://github.com/openshift/origin/pull/4978#issuecomment-146021261

PI-Victor replied:
:ok_hand:  working on it.

PI-Victor commented:
so here's a summary of the changes and what it impacted:
for the virtualbox provider the pre-provisioning script in contrib/vagrant/provisioning-full.sh was not running thus the PATH was not exported and added to root and vagrant user to include the binaries for openshift.

downside: the vagrant up command takes a lot more since it has to run through yum update and the install of the other tools mentioned in the above script. (5-6 minutes in total) compared to 1 minute without.

second i added the ethtool to yum install for this because kubernetes complains that it needs it when running openshift.
i also updated some of the steps inside the CONTRIBUTING guide and added an export PATH for local builds to include the _output dir for openshift, since the binaries are placed there and not in $GOPATH/bin.

please tell me if i've missed anything or if you find anything out of place.

deads2k commented on CONTRIBUTING.adoc:
```diff
@@ -86,13 +90,10 @@ export PATH=$PATH:$GOPATH/bin
 
         $ sudo chmod +r openshift.local.config/master/openshift-registry.kubeconfig
-        $ sudo chmod +r openshift.local.config/master/admin.kubeconfig
+        $ sudo chmod a+rw openshift.local.config/master/admin.kubeconfig
```
Which command attempting to write this file?  I wouldn't expect this to change after its been created.

PI-Victor replied:
after you run openshift with sudo and try to login or create a project from the vagrant user, this is what happens:
`
[vagrant@openshiftdev ~]$ oc login
Error loading config file "/home/vagrant/openshift.local.config/master/admin.kubeconfig": open /home/vagrant/openshift.local.config/master/admin.kubeconfig: permission denied
[vagrant@openshiftdev ~]$ ls -ltr /home/vagrant/openshift.local.config/master/admin.kubeconfig
-rw-------. 1 root root 7300 Oct 15 17:55 /home/vagrant/openshift.local.config/master/admin.kubeconfig
[vagrant@openshiftdev ~]$ sudo chmod g=r,o=r /home/vagrant/openshift.local.config/master/admin.kubeconfig
[vagrant@openshiftdev ~]$ ls -ltr /home/vagrant/openshift.local.config/master/admin.kubeconfig
-rw-r--r--. 1 root root 7300 Oct 15 17:55 /home/vagrant/openshift.local.config/master/admin.kubeconfig
[vagrant@openshiftdev ~]$ oc login
Authentication required for https://10.0.2.15:8443 (openshift)
Username: test
Password: 
Login successful.

You don't have any projects. You can try to create a new project, by running

    $ oc new-project <projectname>

error: open /home/vagrant/openshift.local.config/master/admin.kubeconfig: permission denied
[vagrant@openshiftdev ~]$ oc new-project test
error: open /home/vagrant/openshift.local.config/master/admin.kubeconfig: permission denied

i don't know exactly what happens but i will try to fix this in another pr.

deads2k replied:
@liggitt who maintains the vision for vagrant?  Looks like they're setting `KUBECONFIG` to point to the generated cluster-admin cert, that ought to be updated.  



> Error loading config file "/home/vagrant/openshift.local.config/master/admin.kubeconfig": open /home/vagrant/openshift.local.config/master/admin.kubeconfig: permission denied

liggitt replied:
We do that in a few places:

images/origin/Dockerfile:
ENV KUBECONFIG /var/lib/openshift/openshift.local.config/master/admin.kubeconfig

vagrant-openshift/action/install_openshift.rb:
export KUBECONFIG=/openshift.local.config/master/admin.kubeconfig

contrib/vagrant/provision-util.sh:
local path="${config_root}/openshift.local.config/master/admin.kubeconfig"
echo "export KUBECONFIG=${path}" >> "${target}"

The installer method of copying the generated kubeconfig to a home directory could work, but would require generating the config first, which vagrant isn't doing at the moment.

I am not sure of all the things expecting to log into vagrant images and immediately have cluster admin permissions. @danmcp, any guidance on this?

danmcp replied:
@liggitt I don't think we are in the habit of having admin permissions setup on the inst images by default.  We have been toying with the idea of a bootstrap image that does do that sort of thing but it hasn't really been used or completed.

PI-Victor commented:
i removed yum update from the provision_full.sh since it opens a whole can of worms for the virtualbox vboxfs and breaks the vbox service. also the rpmdb for the fedora image we're currently running is broken and causes issues. this pull ensure we're running the latest docker and 1.5.1 go, since the ones that comes with the standard fedora image are docker 1.6 and go 1.4.2

danmcp commented on Vagrantfile:
```diff
@@ -227,6 +227,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
       v.memory            = vagrant_openshift_config['memory'].to_i
       v.cpus              = vagrant_openshift_config['cpus'].to_i
       v.customize ["modifyvm", :id, "--cpus", vagrant_openshift_config['cpus'].to_s]
+      full_provision(override.vm)
```
I really don't want to add this step.  And I am guessing bigger changes would be necessary to make this the right change.  In general we have taken the mindset of keeping the images up to date rather than running provision on each vagrant up call.

PI-Victor replied:
completely agree. the problem was this image is really outdated and has some issues with being in sync with the docs. PATH is not exported and so on, so basically people kept modifying the docs with ./oadm or ./openshift.
there was a talk yesterday and today about switching to a fedora 23 or centos 7 image for vagrant, what do you think?

danmcp replied:
We already have centos7.  Just change os on origin-init.

I am updating the images now (still fedora 21 but running all the updates).  Let's see what shape we are in after that.

rhcarvalho commented on contrib/vagrant/provision-full.sh:
```diff
@@ -3,8 +3,9 @@ set -euo pipefail
 
-yum update -y
-yum install -y docker-io git vim golang e2fsprogs tmux httpie ctags hg bind-utils which
+#yum update for fedora provider in vagrant breaks vboxfs 
```
Sometimes I run `yum update` manually. The tip I can leave here is to use the vagrant plugin [vbguest](https://github.com/dotless-de/vagrant-vbguest) that will automatically upgrade/recompile the Guest Additions in the vm.

I agree with @danmcp that we should rather have a more up-to-date image than running this script on `vagrant up`.

rhcarvalho commented on contrib/vagrant/provision-full.sh:
```diff
@@ -3,8 +3,9 @@ set -euo pipefail
-yum install -y docker-io git vim golang e2fsprogs tmux httpie ctags hg bind-utils which
+#yum update for fedora provider in vagrant breaks vboxfs 
+#yum update -y
+yum install -y docker-io git vim golang e2fsprogs tmux httpie ctags hg bind-utils which ethtool
```
`ethtool` (and probably all others?) should be part of the base image, @kargakis [added it there](https://github.com/openshift/vagrant-openshift/pull/332) a few weeks ago, we just need to have the base image updated.

danmcp replied:
New images have been published for centos/fedora for virtualbox and libvirt.  Please give those a shot.

@rhcarvalho And you are correct that @kargakis's change should now be in there.

danmcp replied:
Note: Make sure to run:

vagrant box remove fedora_inst
or
vagrant box remove centos7_inst

to pick up the new images.

PI-Victor commented:
thanks for the help. closing this pr since it's not needed.
